tools/src/test/java/org/apache/kafka/tools/GetOffsetShellTest.java

```diff
@@ -48,7 +53,7 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
 @ExtendWith(value = ClusterTestExtensions.class)
-@ClusterTestDefaults(clusterType = Type.ZK)
+@ClusterTestDefaults(clusterType = Type.ALL)
 @Tag("integration")
```

Comment:
Could you use `serverProperties` to define default server configs?
```java
        cluster.config().serverProperties().put("auto.create.topics.enable", false);
        cluster.config().serverProperties().put("offsets.topic.replication.factor", "1");
        cluster.config().serverProperties().put("offsets.topic.num.partitions", String.valueOf(offsetTopicPartitionCount));
```
could be replaced by 
```java
@ClusterTestDefaults(clusterType = Type.ALL, serverProperties = {
    @ClusterConfigProperty(key = "auto.create.topics.enable", value = "false"),
    @ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1"),
    @ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "4")
})
```

Reply:
those `serverProperties` are annotated with `@BeforeEach`. Do you mean you want to put them like below ?

```
@ClusterTest(clusterType = Type.ALL, serverProperties = {
            @ClusterConfigProperty(key = "auto.create.topics.enable", value = "false"),
            @ClusterConfigProperty(key = "offsets.topic.replication.factor", value = "1"),
            @ClusterConfigProperty(key = "offsets.topic.num.partitions", value = "4")
    })
```

If so we'd have to modify each test. `ClusterTestDefaults` didn't have the API `serverProperties` so I think it wouldn't compile

Reply:
> ClusterTestDefaults didn't have the API serverProperties so I think it wouldn't compile

It is supported now. see #15687

Reply:
Thanks! I've pushed a commit to address all your comments.